File: 88.合并两个有序数组.java
/*
 * @lc app=leetcode.cn id=88 lang=java
 *
 * [88] 合并两个有序数组
 */

// @lc code=start
class Solution {
    public void merge(int[] nums1, int m, int[] nums2, int n) {
        int ptr1 = m - 1, ptr2 = n - 1, ptr = m + n - 1;
        while (ptr1 >= 0 && ptr2 >= 0) {
            if (nums1[ptr1] > nums2[ptr2]) {
                nums1[ptr--] = nums1[ptr1--];
            } else {
                nums1[ptr--] = nums2[ptr2--];
            }
        }

        while (ptr1 >= 0)  nums1[ptr--] = nums1[ptr1--];
        while (ptr2 >= 0)  nums1[ptr--] = nums2[ptr2--];
    }
}
// @lc code=end

File: 88.合并两个有序数组.py
#
# @lc app=leetcode.cn id=88 lang=python3
#
# [88] 合并两个有序数组
#

# @lc code=start
class Solution:
    def merge(self, nums1: List[int], m: int, nums2: List[int], n: int) -> None:
        """
        Do not return anything, modify nums1 in-place instead.
        """
        tail = m + n - 1
        m -= 1
        n -= 1
        while m >= 0 and n >= 0:
            if nums1[m] > nums2[n]:
                nums1[tail] = nums1[m]
                m -= 1
            else:
                nums1[tail] = nums2[n]
                n -= 1
            tail -= 1
        while n >= 0:
            # finished copying nums1 but not nums2
            nums1[tail] = nums2[n]
            n -= 1
            tail -= 1


# @lc code=end

